Frontend UI & API calls to few Spotify end-points

tejasraibagi26 opened the pull request:
# Frontend UI

### Additions
- ## Home Page
  - ### Desktop View <img width="1552" alt="Screen Shot 2022-03-23 at 9 15 22 PM" src="https://user-images.githubusercontent.com/60872437/159822905-6c54f328-7b7c-4af5-a8be-d443236b724c.png">
  - ### Tablet View <img width="1552" alt="Screen Shot 2022-03-23 at 9 15 36 PM" src="https://user-images.githubusercontent.com/60872437/159822994-67f0d87e-b77f-4a90-9556-2993328c3196.png">
  - Mobile View (Navbar not responsive yet) <img width="1552" alt="Screen Shot 2022-03-23 at 9 15 44 PM" src="https://user-images.githubusercontent.com/60872437/159823064-019ef589-abd4-42eb-b621-431f1c0bc08f.png">
- ## Login with Spotify
  - Currently won't work (requires server side callback)

tejasraibagi26 committed:
Frontend UI & API calls to few spotify end-points

troytomasch commented:
A couple notes
- Could we add a little more padding between the scroll bars and the text above them?
- If the window is too short then I'm unable to go down to the categories scroll bar
- For the messages sidebar -
- I'm not sure if "Search chat" is the correct wording. Does it search for messages or for chat rooms themselves?
- When I hover over the icon for a chat room, the text changes from gray to white, unless that has different functionality from clicking on the rest of the button, I don't think it should change colors

troytomasch commented on src/Components/Messages/Messages.jsx:
```diff
@@ -0,0 +1,46 @@
+import "./message.css";
+
+const Messages = () => {
+  const tempMessagesRoom = [
+    {
+      id: Math.round(Math.random() * 10000000),
+      title: "Pop",
+    },
+    {
+      id: Math.round(Math.random() * 10000000),
+      title: "EDM",
+    },
+    {
+      id: Math.round(Math.random() * 10000000),
+      title: "Electro",
+    },
+    {
+      id: Math.round(Math.random() * 10000000),
+      title: "Chill",
+    },
+  ];
```
Will this be populated with the message rooms from the database?

tejasraibagi26 replied:
Yes, these are only created as placeholders.

troytomasch reviewed (changes requested):
I left a comment with a few minor cosmetic changes

tejasraibagi26 committed:
Changes

tejasraibagi26 commented:
> A couple notes
> 
> * Could we add a little more padding between the scroll bars and the text above them?
> * If the window is too short then I'm unable to go down to the categories scroll bar
> * For the messages sidebar -
> * I'm not sure if "Search chat" is the correct wording. Does it search for messages or for chat rooms themselves?
> * When I hover over the icon for a chat room, the text changes from gray to white, unless that has different functionality from clicking on the rest of the button, I don't think it should change colors

- Changed placeholder from "Search chat..." to "Search chatroom..."
- Added padding between text and scrollbars
- Chatroom hover icon bug fixed
- Layout not yet perfectly responsive for mobile view.

Sach24eth commented:
All good from my end. Do you want to have the chat room navigation on the same "homepage" as such instead of a separate page?

tejasraibagi26 commented:
> All good from my end. Do you want to have the chat room navigation on the same "homepage" as such instead of a separate page?

It's better to have the chat on the same layout, so the user can chat and browse at the same time.
